external/defects4c/scripts/checkout_bug.py

```
@@ -0,0 +1,132 @@
+    if not repo_url and args.repo_url:
+        repo_url = args.repo_url.strip()
+    if not repo_url:
+        print(f"No repo_url for project '{project}' (not in catalog projects_info). Use --repo-url URL for PROJECT@SHA.", file=sys.stderr)
+        sys.exit(1)
+    if not is_valid_repo_url(repo_url):
+        print("Invalid repo_url: only https:// or git@ URLs are allowed.", file=sys.stderr)
+        sys.exit(1)
+
+    if not os.path.isdir(os.path.join(project_dir, ".git")):
+        print(f"Cloning {project}...")
+        subprocess.run(["git", "clone", repo_url, project_dir], check=True, cwd=work_dir)
```

Ruff (0.15.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)